src/com/maddyhome/idea/vim/helper/SearchHelper.java

```diff
@@ -477,7 +477,7 @@ private static boolean checkInString(@NotNull CharSequence chars, int pos, boole
 
     boolean inString = false;
     boolean inChar = false;
-    for (int i = offset; i <= pos; i++) {
+    for (int i = offset; i <= pos && chars.length()>i; i++) {
```

Comment:
I do this just for a template workaround, I am not understand the original code, if you have fixed this problem, I will revert my commit and checkout your version